File: Data Structures & Algorithms/jump-game-vii/submission-2.cpp
class Solution {
public:
    bool canReach(string s, int minJump, int maxJump) {
        queue<int> bfs;
        bfs.push(0);
        int last_checked = 0;
        int len = s.length();
        while(!bfs.empty()){
            int curr = bfs.front();
            bfs.pop();
            if(curr == len - 1) return true;

            int min_range = max(minJump + curr, last_checked);
            int max_range = min(maxJump + curr, len - 1);
            for(int i = min_range; i <= max_range; ++i){
                if(s[i] == '0') bfs.push(i);
            }

            last_checked = max_range;
        }
        return false;
    }
};

File: Data Structures & Algorithms/insert-new-interval/submission-2.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        for i in range(len(intervals)):
            curr = intervals[i]
            if curr[1] < newInterval[0]:
                res.append(curr)
            elif curr[0] > newInterval[1]:
                res.append(newInterval)
                return res + intervals[i:]
            else:
                newInterval[0] = min(newInterval[0], curr[0])
                newInterval[1] = max(newInterval[1], curr[1])
        res.append(newInterval)    
        return res